datayoga-py/datayoga/jmespath_custom_functions.py

```diff
@@ -41,3 +47,83 @@ def _func_mid(self, element, offset, amount):
     @functions.signature({"types": ["string", "null"], "variadic": True})
     def _func_split(self, element, delimiter=","):
         return str(element).split(delimiter) if element is not None else None
+
+    @functions.signature()
+    def _func_uuid(self):
+        """Generates random uuid4 and returns it as string in standard format."""
+
+        return str(uuid4())
+
+    @functions.signature({"types": ["number", "string", "boolean", "array", "object", "null"], "variadic": True})
+    def _func_hash(self, obj, hash_name="sha1"):
+        """\
+        Calculates hash using given `hash_name` hash function and returns hexadecimal representation.
+
+        The following algorithms were tested:
+
+        - sha256
+        - md5
+        - sha384
+        - sha3_384
+        - blake2b
+        - sha1
+        - sha512
+        - sha3_224
+        - sha224
+        - sha3_256
+        - sha3_512
+        - blake2s
+
+        See https://docs.python.org/3/library/hashlib.html for more information.
+        """
+
+        def prepare() -> Union[bytes, bytearray]:
+            if isinstance(obj, (bytes, bytearray)):
+                return obj
+
+            if obj is None:
+                return b""
+
+            if isinstance(obj, str):
+                return obj.encode()
+
+            return orjson.dumps(obj)
+
+        h = hashlib.new(hash_name)
+        h.update(prepare())
+        return h.hexdigest()
+
+    @functions.signature({"types": ["string", "number"]})
+    def _func_time_delta_days(self, dt):
+        """\
+        Returns the number of days between given `dt` and now(positive)
+        or the number of days that have passed from now(negative).
+
+        If `dt` is string ISO datetime(2011-11-04T00:05:23+04:00) is assumed.
+        If `dt` is number unix timestamp is assumed.
```

Comment:
If `dt` is a number,  Unix timestamp is assumed.

Reply:
✅